subsys/testsuite/ztest/src/ztest_new.c

```diff
@@ -464,12 +476,51 @@ struct ztest_unit_test *ztest_get_next_test(const char *suite, struct ztest_unit
 	return NULL;
 }
 
+#if (IS_ENABLED(CONFIG_ZTEST_DO_THE_SHUFFLE))
+static void ztest_seed_the_shuffle()
+{
+	uint32_t z_seed = SHUFFLE_SEED;
+
+	if (z_seed == 0) {
+#if (IS_ENABLED(CONFIG_POSIX_CLOCK))
```

Comment:
I don't think you need this. Looking at `drivers/timer/native_posix_timer.c` there's a definition of `sys_clock_cycle_get_32()` which should mean that you can just always use `k_cycle_get_32()`.

Reply:
In POSIX scenario I tried using k_cycle_get_32 it always returned 0 as this is called during boot-up.  With non-posix builds, this returned different values around the 1000 range.